### DISCUSSION TITLE
RuCore / Профили ресурсов / Core_Patient

### DISCUSSION THREAD

**Discussion by eugenekogan-spb** [fhir-ru/core#235] [Минимальный набор артефактов]
//--------------------------------------
//страница: Руководство RuCore / Профили ресурсов / Core_Patient
//--------------------------------------

```
Profile: Core_Patient
Id:      core-patient
Parent:       Patient
Title: "Пациент (Ru Core)"
Description: "Пациент"

* address ^short = "Адрес пациента"
* address only Core_Address

* identifier ^slicing.discriminator.type = #value
* identifier ^slicing.discriminator.path = "system"
* identifier ^slicing.rules = #open
* identifier ^slicing.description = "Нарезка по типам идентификаторов пациента - Паспорт РФ, ИНН, СНИЛС, номер МК, полис ОМС, свидетельство о рождении, номер пациента в МИС"
//для отладки оставили только свидетельство о рождении
* identifier contains birthCertificate 0..1

* identifier[birthCertificate] ^short = "Государственное свидетельство о рождении"
  * value only string
  * system 1..1
  * system = "http://fhir.ru/core/systems/birth-certificate"
  * type 1..1
  * type ^short = "Тип идентификатора, кодируется по НСИ"
  * type from $Ядро_НаборЗначений_ДокументыУдостоверяющиеЛичность (required)
  * type = $МЗРФ_Справочник_ДокументыУдостоверяющиеЛичность#6

* name ^short = "ФИО пациента"
  * family ^short = "Фамилия пациента"
  * given ^short = "Имя и отчество пациента. Первым должно идти имя, вторым - отчество"
  * use ^short = "Тип имени пациента. Рекомендуемое значение: official"



Instance:   core-patient-instance-full
InstanceOf: Core_Patient
Usage: #example

* address = core-address-instance-full

* birthDate = "1955-01-10"

* identifier[birthCertificate]
  * system = "http://fhir.ru/core/systems/birth-certificate"
  * value = "IIЮБ-123456"
  * type = $МЗРФ_Справочник_ДокументыУдостоверяющиеЛичность#6

* name
  * family = "Габсбург-Лотарингская"
  * given[0] = "Мария-Антуанетта"
  * given[1] = "Иоганновна"
  * use = #official
```

Методические рекомендации по профилированию ресурсов и типов данных: https://github.com/fhir-ru/core/discussions/248

**Comment by eugenekogan-spb** [comment]
Поправлено для отражения комплексного расширения адреса - [238](https://github.com/fhir-ru/core/discussions/238), на основании решений описанных в [242](https://github.com/fhir-ru/core/discussions/242#discussioncomment-7457586).

**Comment by eugenekogan-spb** [comment]
Пример поправлен на основании устного комментария В.Родионова в обсуждении 07.11.23: вместо подробного описания адреса дана ссылка на пример адреса.

**Comment by Vrodvi** [comment]
# Замечание 1.

Лишние записи, которые выглядят как наложение ограничений на кардинальность, но ими не являются:
```
* . 0..*
* address 0..*
```

Рекомендуется убрать.
Или заменить на использование отметки Must Support (если подходит по смыслу - обсуждали тут - https://github.com/fhir-ru/core/discussions/38).

**Reply by eugenekogan-spb** [reply]
Лишние записи убраны.

**Reply by Vrodvi** [reply]
Замечание "1." закрыто.

**Comment by Vrodvi** [comment]
# Замечание 2.

Комментарии к атрибутам не соответствуют их формализованному описанию или примерам экземпляров.

Пример 1:
```
* name.use ^short = "Тип имени пациента.Фиксированное значение 
Fixed value: official"
```
- формализовано никакого фиксированного значения не установлено

Пример 2:
```
* name.given ^short = "Имя и отчество пациента. Первым должно идти имя, вторым - отчество"
```
- в примере при этом указано лишь одно значение, так ещё и в нестандартном виде - с какой-то вертикальной палкой - `* name.given = "Мария-Антуанетта | Иоганновна"`

**Reply by eugenekogan-spb** [reply]
Исправлено.

**Reply by Vrodvi** [reply]
Замечание "2." закрыто.

**Comment by Vrodvi** [comment]
# Замечание 3.

Непоследовательное форматирование ограничений:
3.1) В одних фрагментах использовано описание через иерархическую вложенность, как тут:
```
* identifier[BirthCertificate].type ^short = "Тип идентификатора"
  * coding 1..1
  * coding from $Ядро_НаборЗначений_ДокументыУдостоверяющиеЛичность (required)
    * code = #6
```
,  а далее - уже использовано описание с нотацией через точку, как тут:
```
* name.family 0..1
```

3.2) В верхнем экземпляре использовано разделение атрибутов на группы через добавление пустой строки, а в нижнем экземпляре всё слитно, без пустых строк.

3.3) Наименование срезов (slice) рекомендуется формировать в формате camelCase - `By convention, the slice names SHOULD be lower camelCase` (https://build.fhir.org/ig/HL7/fhir-shorthand/reference.html#step-2-define-the-slices)
```
* identifier contains 
  BirthCertificate 0..1
```

FSH формат является человекочитаемым, и поэтому следует выработать общие стили форматирование описания и придерживаться их в описаниях всех артефактов.

**Reply by eugenekogan-spb** [reply]
3.1, 3.2. Приведено к единому виду. В предложенном варианте: атрибуты верхнего уровня разделены пустой строкой, атрибуты следующего уровня описаны "лесенкой".
3.3. Поправлено. Рекомендация внесена в заготовку Методических рекомендаций по написанию профилей ресурсов и типов данных.

**Reply by Vrodvi** [reply]
Замечание "3." закрыто.

**Comment by Vrodvi** [comment]
# Замечание 4.

Что за пространство идентификаторов использовано для указания номеров свидетельство о рождении - `* identifier[BirthCertificate].system = "http://fhir.ru/core/systems/birth-certificate"`?
Описания данного пространства нет в минимальном наборе артефактов - необходимо добавить.

Также есть ряд вопросов по идентификации самого пространства номеров свидетельств о рождении -  `http://fhir.ru/core/systems/birth-certificate`:
- почему использован URL, а не URN, как в случае с пространством для ФИАС?
- почему в URL указан `http://fhir.ru/core/...` - сообщество FHIR-RU берёт на себя ответственность за ведение этого пространства? Нет минздравовского идентификатора для этих целей и если нет, то не следует ли предложить его завести?
  - в СЭМД'е свидетельства о рождении, к примеру, не использовали номер свидетельства как идентификатор, а он указывается просто в виде пар атрибут-значение (https://portal.egisz.rosminzdrav.ru/materials/4059)
- есть гарантии того, что комбинации серия-номер свидетельства не переиспользуются со временем? В рамках пространства должна гарантироваться уникальность идентификаторов. Если комбинации переиспользуются, то необходимы дополнительные меры по обеспечения уникальности - к примеру, добавление интервала действия данного идентификатора через атрибут period, но этого не сделано.
- почему тип идентификатора обозначен как необязательный - `* identifier[BirthCertificate].type 0..1` - если далее он устанавливается в фиксированное значение, тут явное противоречие.
(также в примере экземпляра для этого атрибута не заполнено значение)

**Reply by eugenekogan-spb** [reply]
1) Описание пространства в минимальном наборе присутствовало, но содержало ошибку в названии - исправлено.
2) url использован согласно правилу, что RuCore содержит страницу для каждой системы именования. При этом мы пространство не ведем, а только описываем.
3) Государство (система ЗАГС) берет на себя заботу об уникальности, но не сообщает, как именно оно это обеспечивает. В частности, нет информации об ограничении периода действия номера. Практически медицинские системы используют серию и номер свидетельства о рождении как идентификатор, в частности, для внутрирегиональной передачи медицинских данных. 
4) Необязательность типа исправлена, в примере указано значение.

**Reply by Vrodvi** [reply]
Пояснения по ведению пространства и гарантиям уникальности в https://github.com/fhir-ru/core/discussions/247 не описаны, там будет заведено соответствующее замечание.
Данное замечание можно закрывать.

Замечание "4." закрыто.

**Comment by Vrodvi** [comment]
# Замечание 5. 

Наименования артефактов - не соответствуют соглашениям (https://fhir.ru/support.fhir-guide.conventions#:naming-systems-naming).
Примеры:
```
Instance: CorePatient-instance-2
```
(также сам идентификатор - бессмысленный - почему в нём указан двойка, если нет примера с единицей? Почему использовано числовое обозначение, а не текстовое, к примеру `instance-full`?)

```
* address = CoreAddress-instance-2
```
(аналогично - название не по соглашению, необоснованное указание двойки при отсутствии примера с единицей)

**Reply by eugenekogan-spb** [reply]
Поправлено по соглашению на UpperCamelCase
Внесено в заготовку метрекомендаций.

**Reply by Vrodvi** [reply]
Для экземпляров Instance - это идентификатор, так что для него нужно использовать соглашения, касающиеся id, а не name,

**Reply by eugenekogan-spb** [reply]
Исправлено в тексте Patient, Address и в рекомендациях.

**Reply by Vrodvi** [reply]
Замечание "5." закрыто.

**Comment by Vrodvi** [comment]
# Замечание 6.

6.1) Указание требования на фиксированное значение описано избыточно:
```
  * coding from $Ядро_НаборЗначений_ДокументыУдостоверяющиеЛичность (required)
    * code = #6
    * display = "Код свидетельства о рождении по НСИ"
    * system = "http://fhir.ru/core/CodeSystem/nsi-identity-document"
```
, должно быть:
```
* identifier[BirthCertificate].type = $МЗРФ_Справочник_ДокументыУдостоверяющиеЛичность#6
```

6.2) Указанный в этом описании идентификатор справочника (`* system = "http://fhir.ru/core/CodeSystem/nsi-identity-document"`) не соответствует описанию самого справочника (https://github.com/fhir-ru/core/discussions/241), где в качестве предпочтительного идентификатора - указан:
```
* uniqueId[3].type = #oid
* uniqueId[3].preferred = true
* uniqueId[3].value = "http://fhir.ru/core/cs/core-cs-nsi-identity-document"
```

**Reply by eugenekogan-spb** [reply]
6.1 не удается перейти на предложенную сокращенную нотацию, так как возникает ошибка при компиляции примера:
error Cannot assign "http://fhir.ru/core/CodeSystem/nsi-identity-document" to this element; a different uri is already assigned: "http://example.org/CodeSystem/nsi-identity-document".

6.2 разночтение устранено.

**Reply by Vrodvi** [reply]
`http://example.org/` указывается когда явно не указан базовый url спецификации при генерации.
Похоже на какой-то технический баг при генерации.
Тут явно нужно приглашать Тимура для анализа процесса генерации.

**Reply by Vrodvi** [reply]
> 6.1 не удается перейти на предложенную сокращенную нотацию,

Вот картинка с успешной обработкой данной нотации - она работает как при описании профиля, так и в экземпляре:
![image](https://github.com/fhir-ru/core/assets/92928137/084adb24-cff7-4fbd-9e79-a1bf24498808)

Рекомендуется перепроверить и перейти на сокращённую нотацию.

**Reply by eugenekogan-spb** [reply]
Указание http://fhir.ru/core в качестве базового пути в конфигурации fsh online решило проблему. 

Текст переведен на сокращенную нотацию. Внесено дополнение в рекомендации.

**Reply by eugenekogan-spb** [reply]
Вопрос 1. 
Сокращенная нотация позволяет увидеть, что указание в этом месте набора значений Nsi_Vs_identity_documents было искусственным, поскольку в данном срезе используется конкретное значение из справочника. 
Теперь тип артефакта "набор значений" не задействован. Надо найти ему применение.

Вопрос 2. При переходе на сокращенную нотацию осталось в силе следующее требование ?
```
* identifier[birthCertificate].code.coding 1..1 
```

**Reply by Vrodvi** [reply]
> Вопрос 1. 

Можно оставить и короткий формат фиксированного значения и обозначение на использование набора значений:
```
  * type from $Ядро_НаборЗначений_ДокументыУдостоверяющиеЛичность (required)
  * type = $МЗРФ_Справочник_ДокументыУдостоверяющиеЛичность#6
```

> Вопрос 2.

`* identifier[birthCertificate].code.coding 1..1` - это требование лишнее, т.к. на самом деле требуется фиксированное значение. При указании значения невозможно не указать coding.

**Reply by eugenekogan-spb** [reply]
Добавлено требование об использовании набора значений. На мой взгляд, бессмысленное, но для примера годится.

**Reply by Vrodvi** [reply]
Замечание "6." закрыто.

**Comment by Vrodvi** [comment]
# Замечание 7.

Для проверки всех возможностей инструментария SUSHI/Publisher рекомендуется добавить к формализованному описанию профиля дополнительные текстовые описания:
- текст введения (заголовочная часть страницы с описанием профиля)
- текст пояснений/вспомогательной информации (подвальная часть страницы с описанием профиля)

Если обозначенные текстовые части добавить в виде Markdown файлов с наименованием файла, сформированного фасету из вот этого описания - см. - https://fshschool.org/docs/sushi/project/#using-the-hl7-ig-publisher-and-auto-builder, подкладывать необходимо в папку `input/pagecontent/`, то они будут подхвачены инструментом SUSHI/Publisher и добавлены в виде текстовок к соответствующим страницам с описанием профилей. 

Фасеты для наименования файлов с описаниями:
- текст введения, фасет - `{artifact-file-name}-intro.md`*
- текст пояснений/вспомогательной информации - `{artifact-file-name}-notes.md`*

\* `{artifact-file-name}` - это `StructureDefinition-{profile-id}`,
например: для профиля пациента это будут файлы: 
- `StructureDefinition-core-patient-intro.md`, рекомендуемое наполнение:
```
# Пациент (Ru Core)

## "Введение"

Профиль описывает человека, получающего медицинскую помощь.
```
- `StructureDefinition-core-patient-notes.md`, рекомендуемое наполнение:
```
## "Полезные ссылки"

- [Страница профиля на Zendoc](https://fhir.ru/core.profile.Patient)
- [Страница методических рекомендаций по формированию профиля](https://fhir.ru/core.profile.Patient)
```

Также просьба рассмотреть возможность разметки подобных текстовых описаний на странице Zendoc, чтобы они автоматически переносились на стандартный сайт руководства по реализации (через предварительное формирование файлов и подкладывание генератору сайта).

**Reply by Vrodvi** [reply]
@eugenekogan-spb Можно ли передать данное замечание Тимуру на проработку?

**Reply by Vrodvi** [reply]
В папке проекта SUSHI должно выглядить примерно так:
![image](https://github.com/fhir-ru/core/assets/92928137/bac8241a-5485-4a09-a6ea-1ac1df340e39)
